src/Testcontainers.Xunit/ContainerTest.cs

```diff
@@ -0,0 +1,16 @@
+namespace Testcontainers.Xunit;
+
+/// <summary>
+/// Base class for tests needing a container per test method.
+/// A logger is automatically configured to write messages to xUnit's <see cref="ITestOutputHelper" />.
+/// </summary>
+/// <typeparam name="TBuilderEntity">The builder entity.</typeparam>
+/// <typeparam name="TContainerEntity">The container entity.</typeparam>
+[PublicAPI]
+public abstract class ContainerTest<TBuilderEntity, TContainerEntity>(ITestOutputHelper testOutputHelper, Func<TBuilderEntity, TBuilderEntity> configure = null)
```

Comment:
Why do we provide the `configure` overload here? Wouldn't it be sufficient to expect developers to override the member?

Reply:
I initially wrote it like that for no particular reason. We can probably get rid of it, if only for consistency with how it's done with `ContainerFixture`.

Reply:
Wait, I take that back, it's pretty convenient actually. For example:

```csharp
public abstract class MongoDbContainerTest(ITestOutputHelper testOutputHelper, Func<MongoDbBuilder, MongoDbBuilder> configure = null)
    : ContainerTest<MongoDbBuilder, MongoDbContainer>(testOutputHelper, configure)
{
    [Fact]
    public void Test()
    {
    }

    [UsedImplicitly]
    public sealed class MongoDbDefaultConfiguration(ITestOutputHelper testOutputHelper)
        : MongoDbContainerTest(testOutputHelper);

    [UsedImplicitly]
    public sealed class MongoDbNoAuthConfiguration(ITestOutputHelper testOutputHelper)
        : MongoDbContainerTest(testOutputHelper, builder => builder.WithUsername(string.Empty).WithPassword(string.Empty));

    [UsedImplicitly]
    public sealed class MongoDbV5Configuration(ITestOutputHelper testOutputHelper)
        : MongoDbContainerTest(testOutputHelper, builder => builder.WithImage("mongo:5.0"));

    [UsedImplicitly]
    public sealed class MongoDbV4Configuration(ITestOutputHelper testOutputHelper)
        : MongoDbContainerTest(testOutputHelper, builder => builder.WithImage("mongo:4.4"));
}
```

Reply:
This is probably an interesting use case, indeed. We should document it somehow, otherwise no one will know about the overload.